### PR TITLE
Lindorm handler implemented. 

### LINKED ISSUE #7887: [Integration] Integrate Lindorm with MindsDB
The goal of this issue is to build a new handler that will connect MindsDB to Lindorm database from Alibaba.

## Resources :bulb: 

:checkered_flag: To get started, please refer to the following resources:

* :books:  [Building Data Handler docs](https://docs.mindsdb.com/contribute/data-handlers): This documentation will guide you through the process of building a custom handler for MindsDB. It contains essential information on the handler structure, methods, and best practices.

* :book:  [Lindorm database from Alibaba](https://www.alibabacloud.com/blog/lindorm-alibaba-clouds-newest-cloud-native-multi-model-database_596269)

* :point_right: [ChromaDB](https://github.com/mindsdb/mindsdb/tree/staging/mindsdb/integrations/handlers/chromadb_handler): This existing handler connects MindsDB with ChromaDB.

* You can find more examples of MindsDB's data handlers [here](https://github.com/mindsdb/mindsdb/tree/staging/mindsdb/integrations/handlers).

## Next Steps :male_detective: :female_detective: 
 
-   You can familiarize yourself with Lindorm.
-   Create a new handler that connects MindsDB to Lindorm.
-   Implement the core features mentioned above.
-   Thoroughly test the integration to ensure proper functionality.
-   Submit a pull request with the new Lindorm Integration handler.

**The PR should include the implementation of the Lindorm handler and unit tests.**

## :loudspeaker: Additional rewards :trophy:

Each data integration brings :two: :five:  : points for SWAG and entry into the draw for a  :point_down: :
-  :computer:  [Razer Blade 16 Laptop](https://www.razer.com/gaming-laptops/Razer-Blade-16/RZ09-0483SEJ3-R3U1)
- :shirt: :bear:  [MindsDB Swag](https://store.covver.io/mindsdb/collections/swag-collection)
:information_source: For more info check out https://mindsdb.com/hacktoberfest/ :point_left:

### PR DESCRIPTION
**Pull request by someshfengde**
## Description
Implemeted the lindorm handler for lindormdb. 

Fixes #7887 

## Type of change
- [x] ⚡ New feature (non-breaking change which adds functionality)
- [x] 📄 This change requires a documentation update

## Additional Media:

https://github.com/mindsdb/mindsdb/assets/42097653/8cd7281c-34bd-4b57-b812-b82d9b25d431

- [x] I have attached a brief loom video or screenshots showcasing the new functionality or change.

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have appropriately commented on my code, especially in complex areas.
- [x] Necessary documentation updates are either made or tracked in issues.
- [x] Relevant unit and integration tests are updated or added.

### CODE REVIEW

**Comment by someshfengde** [comment]
Lindor database is slow af!! 😯 
This will be the first and last time I've used alicloud 😢

**Comment by martyna-mindsdb** [comment]
@someshfengde 
In the video you provided, you create a connection. Please test querying a table from this database.

**Comment by someshfengde** [comment]
Hi @martyna-mindsdb I didn't show that in video because ah there was not any way to update the database through python over there. The service is painfully slow 😭 even just starting that damn lindorm instance takes about an hour. ( every configuration , making public link, connecting with the database requires update the setting of instance ( each thing takes about 10 minutes ) 

Sorryy can't check this 😢

**Comment by martyna-mindsdb** [comment]
Hi @someshfengde 
Did you manage to query a table from this Lindorm database?

Also, please ensure that the code_checking passes above.

**Comment by someshfengde** [comment]
Ohh lindorm instance didn't have any data in their database by default. That's why querying from that database gave error Even I tried to pass the data into that but that wasn't working ( not even with the normal phoenixdb library. There is not enough documentation for doing the same :(

**Comment by someshfengde** [comment]
I'll check on the linting issues

**Comment by someshfengde** [comment]
Hi @martyna-mindsdb  I've updated the code for fixing the code_checking after code checking there is still this tests for deploying over to ubuntu fails 

is this supposed to be like this ? 
Screenshot 2023-10-23 at 22 34 32

File: mindsdb/integrations/handlers/lindorm_handler/lindorm_handler.py [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+        Returns:
+            HandlerStatusResponse
+        """
+
+        if self.is_connected is True:
+            return self.connection
+
+        lindorm_connection_data = {'lindorm_user': self.connection_data['lindorm_user'], 'lindorm_password': self.connection_data['lindorm_password']}
+
+        print(self.connection_data['url'])
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/lindorm_handler/lindorm_handler.py:64]: Can you remove this?

File: mindsdb/integrations/handlers/lindorm_handler/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+pyphoenix
+phoenixdb
+protobuf==3.20.*
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/lindorm_handler/requirements.txt:3]: Is protobuf required here?

File: mindsdb/integrations/handlers/lindorm_handler/lindorm_handler.py [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+            )
+
+        cursor.close()
+        if need_to_close is True:
+            self.disconnect()
+
+        return response
+
+
+connection_args = OrderedDict(
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/lindorm_handler/lindorm_handler.py:224]: Let's add all of the bellow keys to the README

File: mindsdb/integrations/handlers/lindorm_handler/lindorm_handler.py [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+    },
+    authentication={
+        'type': ARG_TYPE.STR,
+        'description': 'An alternative way to specify the authentication mechanism that mimics the semantics of the JDBC drirver.'
+    },
+    avatica_user={
+        'type': ARG_TYPE.STR,
+        'description': 'The username for BASIC or DIGEST authentication. Use in conjunction with the authentication option.'
+    },
+    avatica_password={
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/lindorm_handler/lindorm_handler.py:249]: This is not used in the code above? Do we need it?

File: mindsdb/integrations/handlers/lindorm_handler/__init__.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+    import_error = None
+except Exception as e:
+    Handler = None
+    import_error = e
+
+title = "Lindorm"
+name = "lindorm"
+type = HANDLER_TYPE.DATA
+
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/lindorm_handler/__init__.py:17]: Can we add the icon here? Please check other handlers

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]

File: mindsdb/integrations/handlers/lindorm_handler/requirements.txt (hunk 2 above)
Comment (someshfengde) [reply mindsdb/integrations/handlers/lindorm_handler/requirements.txt:3]: Hi @ZoranPandovski it was using same requirements for the phoenixdb handler. It was there present in that handler's requirements (for phoenixdb) that's why I've put it there .

File: mindsdb/integrations/handlers/lindorm_handler/__init__.py (hunk 5 above)
Comment (someshfengde) [reply mindsdb/integrations/handlers/lindorm_handler/__init__.py:18]: icon for lindormdb /

File: mindsdb/integrations/handlers/lindorm_handler/lindorm_handler.py (hunk 4 above)
Comment (someshfengde) [reply mindsdb/integrations/handlers/lindorm_handler/lindorm_handler.py]: Nope we probably don't it was present in the phoenixdb this is simply copy of that just with the lindorm_db_user and lindorm_password

File: mindsdb/integrations/handlers/lindorm_handler/__init__.py (hunk 5 above)
Comment (someshfengde) [reply mindsdb/integrations/handlers/lindorm_handler/__init__.py:18]: added into latest. commit

File: mindsdb/integrations/handlers/lindorm_handler/lindorm_handler.py (hunk 4 above)
Comment (someshfengde) [reply mindsdb/integrations/handlers/lindorm_handler/lindorm_handler.py]: Removed the code.

File: mindsdb/integrations/handlers/lindorm_handler/lindorm_handler.py (hunk 3 above)
Comment (someshfengde) [reply mindsdb/integrations/handlers/lindorm_handler/lindorm_handler.py]: added the inputs into README in latest commit

File: mindsdb/integrations/handlers/lindorm_handler/lindorm_handler.py (hunk 1 above)
Comment (someshfengde) [reply mindsdb/integrations/handlers/lindorm_handler/lindorm_handler.py]: Removed this.

**Review by ZoranPandovski — APPROVED** [review]

**Pull request merged**